src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.h

```diff
@@ -8,13 +8,141 @@ struct BossGanon;
+    /* 10 */ GDF_FWORK_MAX
+} GanondorfFwork;
+
+typedef enum {
+    /* 0 */ GDF_WINDOW_SHATTER_OFF,
+    /* 1 */ GDF_WINDOW_SHATTER_PARTIAL,
+    /* 2 */ GDF_WINDOW_SHATTER_FULL
+} WindowShatterState;
+
+typedef enum {
+    /* 0 */ GDF_SHOCK_DORF_YELLOW,
+    /* 1 */ GDF_SHOCK_PLAYER_YELLOW,
+    /* 2 */ GDF_SHOCK_PLAYER_PURPLE
+} GanondorfShockType;
+
+typedef enum {
+    /* 0 */ GDF_EFF_NONE,
+    /* 1 */ GDF_EFF_SPARKLE,
+    /* 2 */ GDF_EFF_LIGHT_RAY,
+    /* 3 */ GDF_EFF_SHOCK,
+    /* 4 */ GDF_EFF_LIGHTNING,
+    /* 5 */ GDF_EFF_IMPACT_DUST_DARK,
+    /* 6 */ GDF_EFF_IMPACT_DUST_LIGHT,
+    /* 7 */ GDF_EFF_SHOCKWAVE,
+    /* 8 */ GDF_EFF_BLACK_DOT,
+    /* 9 */ GDF_EFF_WINDOW_SHARD
+} GanondorfEffectType;
+
+typedef enum {
+    /* 0 */ GDF_TRIFORCE_PLAYER,
+    /* 1 */ GDF_TRIFORCE_ZELDA,
+    /* 2 */ GDF_TRIFORCE_DORF
+} GanondorfTriforceType;
+
 typedef struct BossGanon {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x4C];
+    /* 0x014C */ s32 animBankIndex;
+    /* 0x0150 */ SkelAnime skelAnime;
+    /* 0x0194 */ BossGanonActionFunc actionFunc;
     /* 0x0198 */ u8 unk_198;
-    /* 0x0199 */ char unk_199[0x57F];
-    /* 0x0718 */ s16 organFadeTimer;
-    /* 0x071A */ char unk_71A[0x2];
-} BossGanon; // size = 0x071C
+    /* 0x0199 */ u8 legSwayEnabled;
+    /* 0x019A */ u8 unk_19A;
+    /* 0x019C */ s16 unk_19C; // timer
+    /* 0x019E */ u8 unk_19E;
+    /* 0x019F */ u8 unk_19F;
+    /* 0x01A0 */ s8 envLightMode;
+    /* 0x01A2 */ s16 unk_1A2;
+    /* 0x01A4 */ s16 unk_1A4;
+    /* 0x01A6 */ s16 unk_1A6;
+    /* 0x01A8 */ s16 unk_1A8;
+    /* 0x01AA */ s16 unk_1AA;
+    /* 0x01AC */ s16 unk_1AC;
+    /* 0x01AE */ s16 triforceType;
+    /* 0x01B0 */ char unk_1B0[0x6];
+    /* 0x01B6 */ s16 timers[5];
+    /* 0x01C0 */ u8 startVolley;
+    /* 0x01C2 */ s16 unk_1C2;
+    /* 0x01C4 */ s16 screenFlashTimer;
+    /* 0x01C8 */ f32 fwork[GDF_FWORK_MAX];
+    /* 0x01F0 */ Vec3f unk_1F0;
+    /* 0x01FC */ Vec3f unk_1FC;
+    /* 0x0208 */ Vec3f unk_208;
+    /* 0x0214 */ Vec3f unk_214;
+    /* 0x0220 */ Vec3f unk_220;
+    /* 0x022C */ Vec3f unk_22C;
+    /* 0x0238 */ Vec3f unk_238;
+    /* 0x0244 */ char unk_244[0x10];
+    /* 0x0254 */ f32 handLightBallScale;
+    /* 0x0258 */ f32 unk_258;
+    /* 0x025C */ u8 unk_25C;
+    /* 0x0260 */ Vec3f unk_260; // hand position?
+    /* 0x026C */ s16 unk_26C; // timer?
+    /* 0x0270 */ f32 unk_270;
+    /* 0x0274 */ u8 unk_274;
+    /* 0x0278 */ Vec3f unk_278;
+    /* 0x0284 */ f32 unk_284; // scale for something
+    /* 0x0288 */ f32 unk_288;
+    /* 0x028C */ f32 unk_28C; // scale for something
+    /* 0x0290 */ f32 unk_290;
+    /* 0x0294 */ f32 unk_294[15];
+    /* 0x02D0 */ f32 unk_2D0; // scale for something
+    /* 0x02D4 */ s16 unk_2D4; // timer
+    /* 0x02D8 */ Vec3f triforcePos;
+    /* 0x02E4 */ u8 shockGlow;
+    /* 0x02E6 */ s16 unk_2E6; // timer
+    /* 0x02E8 */ s16 unk_2E8; // timer
+    /* 0x02EC */ Vec3f bodyPartsPos[15];
+    /* 0x03A0 */ char unk_3A0[0x24];
```

Comment:
what part of `BossGanon_UpdateEffects` are you referring to? the part relevant to ganondorf does `bodyPart = (s16)Rand_ZeroFloat(13.9f) + 1;` while the block for player does `bodyPart = (s16)Rand_ZeroFloat(17.9f);`

Reply:
I think I was looking at the player part so nvm about that.
This should probably still be size 18 like the other body part arrays tho, but I would understand if you don't want to do that

Reply:
i agree with you that it was probably this size, but yea i dont see anything access that high.
additionally it looks like it may only be a body parts pos in the context of ganondorf, while the other instances are using it for something else. so this will probably need a vDefine

Reply:
because of this i am going to make it size 18 and un-name it with a comment for now

Reply:
I see, I guess we'll figure it out later